src/main.ts

```diff
@@ -58,19 +58,43 @@ processor.run(new TypeormDatabase({supportHotBlocks: true}), async (ctx) => {
                 CONTRACTS_TO_INDEX.some(contract => contract.address == log.address) &&
                  log.topics[0] === erc1155.events.TransferSingle.topic
             ) {
-                // TODO Handle ERC1155 TransferSingle event
+                const {from, to, id, value} = erc1155.events.TransferSingle.decode(log);
+                TransfersNfts.push({
+                    id: uuidv4(),
+                    block: log.block,
+                    from: from,
+                    to: to,
+                    tokenId: id,
+                    amount: value,
+                    contractAddress: log.address,
+                    blockchain: BLOCKCHAIN,
+                    contractType: ContractType.erc1155
+                })
             }
             else if(
                 CONTRACTS_TO_INDEX.some(contract => contract.address == log.address) &&
                  log.topics[0] === erc1155.events.TransferBatch.topic
             ) {
-                // TODO handle ERC1155 TransferBatch event
+                const {from, to, ids, amounts} = erc1155.events.TransferBatch.decode(log);
+                for(let i=0; i<ids.length; i++){
```

Comment:
Here if you dont need index, use let x of array

Reply:
I need an index because TransferBatch looks like this: TransferBatch(address operator, address from, address to, uint256[] ids, uint256[] values), so ids and values are correlated, eg: (ids [1, 2], values [10, 20], meaning transfer tokenid 1 of amount 10, transfer tokenid2 of amount 20) so i need to access both ids and values at same index